python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -215,6 +220,107 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info, fil
     return embedding
 
 
+class CategoricalDropout(layers.Layer):
+    """Custom Dropout class for categorical indices"""
+
+    def __init__(self, dropout_rate, seed=None, **kwargs):
+        """
+        Args:
+            dropout_rate: fraction of units to drop, i.e., set to OOV token 0
+            seed: random seed for sampling
+        """
+        super(CategoricalDropout, self).__init__(**kwargs)
+        self.dropout_rate = dropout_rate
+        self.seed = seed
+
+    def get_config(self):
+        config = super(CategoricalDropout, self).get_config()
+        config.update({"dropout_rate": self.dropout_rate, "seed": self.seed})
+        return config
+
+    def call(self, inputs, training=None):
+        # At training time, mask indices to 0 at dropout_rate
```

Comment:
This isn't a trivial operation to understand why we need it.
Could you add a python doc about why we need to do it, and an example of input output?

Reply:
It works similar to this without the scaling part -> https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dropout

The example is in the test, right?

Reply:
I've added an example in the comments now.